common/dtos/websocket_message.go

```diff
@@ -6,19 +6,25 @@ import "encoding/json"
 type WebSocketMessageType int32
 
 const (
-	//WSMsgError indicates the WebSocketMessage is an error response
+	//WSMsgError indicates this WebSocketMessage contains an error object
 	WSMsgError WebSocketMessageType = iota
-	//WSMsgRequestRegisterSlave indicates this is a request for the master to register a new slave
-	WSMsgRequestRegisterSlave
-	//WSMsgResponseRegisterSlave indicates this is a response to a previous request to register a new slave
-	WSMsgResponseRegisterSlave
+	/*WSMsgAuthenticationRequest indicates this WebSocketMessage contains an
+	AuthenticationRequest*/
+	WSMsgAuthenticationRequest
+)
+
+const (
+	/*WSMsgAuthenticationResponse indicates this WebSocketMessage contains an
+	AuthenticationResponse*/
+	WSMsgAuthenticationResponse WebSocketMessageType = iota + 10001
 )
 
 //WebSocketMessage represents a message received from a client or
 //ready to be sent to it
 type WebSocketMessage struct {
-	Type    WebSocketMessageType
-	Payload []byte
+	Type      WebSocketMessageType `json:"type"`
```

Comment:
In my opinion "type" could be misleading name. I suggest renaming it to something more readable, e.g "messageType" or "msgType".